src/main/java/com/pinterest/secor/reader/SecorKafkaMessageIterator.java

```diff
@@ -0,0 +1,166 @@
+        optionalConfig(config.getSaslMechanism(), conf -> props.put("sasl.mechanism", conf));
+        optionalConfig(config.getSecurityProtocol(), conf -> props.put("security.protocol", conf));
+        optionalConfig(config.getSslEnabledProtocol(), conf -> props.put("ssl.enabled.protocols", conf));
+        optionalConfig(config.getSslKeystoreType(), conf -> props.put("ssl.keystore.type", conf));
+        optionalConfig(config.getSslProtocol(), conf -> props.put("ssl.protocol", conf));
+        optionalConfig(config.getSslProvider(), conf -> props.put("ssl.provider", conf));
+        optionalConfig(config.getSslTruststoreType(), conf -> props.put("ssl.truststore.type", conf));
+        optionalConfig(config.getNewConsumerPartitionAssignmentStrategyClass(), conf -> props.put("partition.assignment.strategy", conf));
+
+        String dualCommitEnabled = config.getDualCommitEnabled();
+        String offsetStorage = config.getOffsetsStorage();
+        final boolean skipZookeeperOffsetSeek = offsetStorage.equals("kafka") && dualCommitEnabled.equals("true");
+
+        mZookeeperConnector = new ZookeeperConnector(config);
+        mRecordsBatch = new ArrayDeque<>();
+        mKafkaConsumer = new KafkaConsumer<>(props);
+        ConsumerRebalanceListener reBalanceListener = new ConsumerRebalanceListener() {
+            @Override
+            public void onPartitionsRevoked(Collection<TopicPartition> assignedPartitions) {
+                for (TopicPartition topicPartition : assignedPartitions) {
+                    LOG.debug("re-balance will happen for assigned topic partition {}", topicPartition);
+                }
+            }
+
+            @Override
+            public void onPartitionsAssigned(Collection<TopicPartition> collection) {
+                if (skipZookeeperOffsetSeek) {
+                    LOG.debug("offset storage set to kafka. Skipping reading offsets from zookeeper");
+                    return;
+                }
+                Map<TopicPartition, Long> committedOffsets = getCommittedOffsets(collection);
+                committedOffsets.forEach(((topicPartition, offset) -> {
+                    if (offset == -1) {
+                        if (offsetResetConfig.equals("earliest")) {
+                            mKafkaConsumer.seekToBeginning(Collections.singleton(topicPartition));
+                        } else if (offsetResetConfig.equals("latest")) {
+                            mKafkaConsumer.seekToEnd(Collections.singleton(topicPartition));
+                        }
+                    } else {
+                        LOG.debug("Seeking {} to offset {}", topicPartition, Math.max(0, offset));
+                        mKafkaConsumer.seek(topicPartition, Math.max(0, offset));
+                    }
+                }));
+            }
+        };
+
+        String[] subscribeList = config.getKafkaTopicList();
+        if (Strings.isNullOrEmpty(subscribeList[0])) {
```

Comment:
Sorry to weigh in while not being a member of the secor project, but I AM running this branch in order to work with a Kafka cluster that includes authentication. This will fail if the user has not specified a topic list.

Reply:
Why the user does not have a topic list?

Reply:
Because they're subscribing based only on topic patterns.

Reply:
This shouldn't break since `kafka.new.consumer.topic.list` is set on `secor.common.properties`. Are you using kubernetes? I forgot to add a bunch configs to the other properties files since they aren't tested.

Reply:
Yeah, this turned out to be an error on my end. Apologies, ignore.